src/engine/routing_algorithms/shortest_path.cpp

```diff
@@ -166,8 +164,7 @@ void search(SearchEngineData<Algorithm> &engine_working_data,
                                     source_phantom.GetReverseWeightPlusOffset(),
                                 source_phantom.reverse_segment_id.id);
         }
-        BOOST_ASSERT(forward_heap.Size() > 0);
```

Comment:
This can happen because we could get `PhantomNodes` that are not admissible as source/target right?